src/clib/pioc_support.c

```diff
@@ -2172,7 +2172,13 @@ int PIOc_openfile_retry(int iosysid, int *ncidp, int *iotype, const char *filena
 
                 /* open netcdf file serially on main task */
                 if (ios->io_rank == 0)
+                {
                     ierr = nc_open(filename, file->mode, &file->fh);
+                    if(ierr != NC_NOERR)
```

Comment:
Should it be "ierr == NC_NOERR" here? Otherwise users will not see the descriptive log as open netcdf file serially most likely will not fail. Also, I find out that PIO_NOERR (wrapper of NC_NOERR) is used in PIO2 code.

Reply:
Yes, i think that has to be ierr == NC_NOERR . I will fix it soon.
Regarding the error code I think we should be using NetCDF error codes when calling NetCDF functions (The rest of the code is not doing the right thing here)

Reply:
Alternatively, we can move this log before calling nc_open, no need to check ierr. This way, no matter nc_open works or not, users are always notified that PIO2 is trying to reopen the file with serial netcdf.